src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java

```diff
@@ -294,6 +304,13 @@ public String getKustoUrl() {
         return this.getString(KUSTO_URL_CONF);
     }
 
+    // TODO [yischoen 2020-09-29]: Next major version bump, just return the value, as it will be a required setting
+    public String getKustoEngineUrl() {
+        if (!Strings.isNullOrEmpty(this.getString(KUSTO_ENGINE_URL_CONF)))
+            return this.getString(KUSTO_ENGINE_URL_CONF);
+        return this.getString(KUSTO_URL_CONF).replaceFirst("(?i)ingest-", "");
```

Comment:
actually while we are at it maybe we should do this using URI and uriBuilder 
this is what we do on the spark connector
```scala
      val host = new URI(cluster).getHost
val startIdx = if (host.startsWith(ingestPrefix)) ingestPrefix.length else 0
        uriBuilder.setScheme("https").setHost(host.substring(startIdx)).toString
```

Reply:
I don't think that would help here. This method is as straightforward as it can get, and should work in all scenarios. Adding it to a URI object first and then unpacking the URL from it a couple lines later at the end of the method would add unnecessary complexity for no benefit to this method. The required object downstream is a String, not a URI.

This logic is even more robust, in that it can handle an ingest URL that doesn't *start* with "ingest-", which was the impetus for this PR. I suggest the Spark connector be updated to use this logic.

Though not here, I do see a benefit for using a URI object in general. Specifically, to verify that the URI is properly formed. However, I think that would best be done using a ConfigDef.Validator when being defined.

Reply:
ok